Guillaume Canneton/week01/strings/js/exercises.js

```diff
@@ -0,0 +1,131 @@
+// Look up the JavaScript string reference to find methods which may be useful!
+// ```
+function mixUp(wordOne, wordTwo){
+  const wordMixOne = wordTwo.slice(0, 2) + wordOne.slice(2);
+  const wordMixTwo = wordOne.slice(0, 2) + wordTwo.slice(2);
+  console.log( wordMixOne + " " + wordMixTwo)
+}
+mixUp("cookie", "doggie")
+
+
+// ## FixStart
+//
+// Create a function called fixStart. It should take a single argument, a string, and return a version where all occurences of its first character have been replaced with '*', except for the first character itself. You can assume that the string is at least one character long. For example:
+// ```
+// fixStart('babble'): 'ba**le'
+// ```
+
+function fixStart(hiddenWord){
+  let first = hiddenWord.slice(0, 1);
+  const replacement = "*";
+  let result = first + replacement.repeat(hiddenWord.length - 1);
+  console.log(result)
+}
+fixStart("Hello")
+// ## Verbing
+//
+// Create a function called verbing. It should take a single argument, a string. If its length is at least 3, it should add 'ing' to its end, unless it already ends in 'ing', in which case it should add 'ly' instead. If the string length is less than 3, it should leave it unchanged. For example:
+// ```
+//   verbing('swim'): 'swimming'
+//   verbing('swimming'): 'swimmingly'
+//   verbing('go'): 'go'
+// ```
+//
+
+function verbing(verb){
+  let result;
+  if (verb.length < 3){
+    result = verb;
+  } else if (verb.slice(verb.length - 3) === "ing"){
+    result = `${verb}ly`;
+  } else {
+    result = `${verb}ing`;
+  }
+  return result;
+}
+console.log(verbing("go"));
+console.log(verbing("going"));
+console.log(verbing("get"));
+
+function verbing2(verb){
+  let result;
+  if (verb.length > 2){
+    if (verb.slice(verb.length - 3) === "ing"){
+      result = `${verb}ly`;
+  } else {
+    result = `${verb}ing`;
+  }
+  } else {
+    result = `${verb}`;
+  }
+  return result;
+}
+console.log(verbing2("swimming"));
+console.log(verbing2("take"));
+console.log(verbing2("go"));
+
+// ## Not Bad
+//
+// Create a function called notBad that takes a single argument, a string.
+// - It should find the first appearance of the substring 'not' and 'bad'.
+// - If the 'bad' follows the 'not', then it should replace the whole 'not'...'bad' substring with 'good' and return the result.
+// - If it doesn't find 'not' and 'bad' in the right sequence (or at all), just return the original sentence.
+//
+// For example:
+// ```
+//   notBad('This dinner is not that bad!'): 'This dinner is good!'
+//   notBad('This movie is not so bad!'): 'This movie is good!'
+//   notBad('This dinner is bad!'): 'This dinner is bad!'
+// ```
+function notBad(sentence){
+  const not = sentence.search("not")
+  const bad = sentence.search("bad")
+  let result;
+  if (not === -1){
+    result = `${sentence}`;
+  } else if (not > bad){
+    result = `${sentence}`;
+  } else {
+    result = `${sentence.slice(0, not)}good${sentence.slice(bad + 2)}`
+  }
```

Comment:
if you add a `return result;` statement at the end of your if/else if/else statement this function works perfectly!